### PR TITLE
Measure message latency and gate recovery

### PR DESCRIPTION
**Pull request by dannym-arx**
Send timing currently starts after dequeue, hiding time spent behind account work. Add aggregate timings for queue wait, local projection, local acceptance and durable fanout preparation, publication, response, and live inbound processing. Host applications can report actual render milestones through Rust, UniFFI, and C; the same metrics reach the existing opt-in OTLP export.

Apps can also report their own stages with `record_host_timing(name, duration, outcome)` through Rust, Swift/Kotlin, and C. Reuse the existing product event registry and consent controls to send registered stage names, duration buckets, and success/failure outcomes to Aptabase. These custom events do not add OTLP series or raw-duration traces. Include a registration example for app-specific layout and decoding measurements.

Add `just test-message-journeys` to check abrupt process exits at four send boundaries, caller cancellation, exact-event retry after unknown outcomes and reconnects, and timeline recovery after subscription overflow. The checks assert recovery without duplicate visible messages or false publication success.

A crash between optimistic projection and durable acceptance still leaves an unconfirmed row without automatic retry. The test and telemetry documentation explicitly preserve that distinction; recovery behavior is unchanged.

### CODE REVIEW

File: crates/marmot-app/src/runtime/commands.rs [hunk 1]
```diff
@@ -1234,17 +1234,25 @@ impl AccountManager {
...
             .await
             .map_err(|_| AppError::TransportClosed)?;
-        let summary = account_worker_response(response).await?;
+        let result = account_worker_response(response).await;
+        self.shared.app_performance_telemetry().record(
+            AppPerformanceOperation::OutboundMessageResponse,
+            enqueued_at.elapsed(),
+            result.is_ok(),
+        );
+        let summary = result?;
```
[the remark this answers is not in the document]
Comment (dannym-arx) [reply crates/marmot-app/src/runtime/commands.rs]: @coderabbitai Fixed in 77d49fee. Both send paths now capture worker lookup, channel admission, and response delivery in the result recorded by OutboundMessageResponse. Added message_journey_early_errors for both paths before and after shutdown; the cancellation regression remains passing. All 8 targeted message journey tests and just fast-ci passed. Please recheck this finding.

File: crates/marmot-app/src/tests/message_journeys.rs [hunk 2]
```diff
@@ -0,0 +1,332 @@
...
+            assert_eq!(
+                message.source_message_id_hex.as_deref(),
+                Some(attempted.id.as_str()),
+                "{phase}: recover the original exact event"
+            );
+            assert!(message.invalidation_status.is_none());
+            let publishes = relay.published_event_ids();
+            assert!(publishes.iter().all(|id| id == &attempted.id));
+            assert!(
+                publishes.len() <= 1,
```
Comment (erskingardner) [inline crates/marmot-app/src/tests/message_journeys.rs]: [P3] Assert the expected publication count for each crash phase

The shared `publishes.len() <= 1` assertion lets the `response` case republish once after reopening, even though the child already received publication confirmation and persisted the final projection. Such a redundant-publication regression would still pass this recovery gate. It also leaves the `durable`/`relay` cases without a positive assertion that recovery actually sent the retained event. Assert zero publications for `response` and exactly one for `durable`/`relay`, retaining the event-ID equality check, so the test distinguishes cleanup-only recovery from an exact-event retry.

**Review by erskingardner — DISMISSED** [review]
| Review metadata | Value |
| --- | --- |
| Reviewed at (UTC) | `2026-09-08T10:04:58Z` |
| Commit reviewed | `77d49fee3a57235e37e4e3a2968526a96f7b6ffb` |
| Model | `gpt-6-astra` |
| Reasoning level | `high` |
| Recommended action | **Merge** |

No blocking implementation issues found. The timing boundaries distinguish optimistic projection, local preparation, exact-event publication evidence, caller response, and actual host rendering. Custom host timings reuse the registered, consent-gated product-event path, and the new snapshot fields and host operations are mirrored through UniFFI and C. The earlier response-admission failure finding is fixed at this head.

One non-blocking inline suggestion strengthens the crash gate by asserting phase-specific publication counts.

Validation used remote source and CI at this exact head; no local checkout changes or test execution. CI logs confirm the crash/reopen, caller-cancellation, overflow, exact-event unknown-outcome retry, telemetry-export, and UniFFI host-metric tests passed. Rust format/check/clippy, WASM, and C smoke checks passed; all three PR commits have verified signatures. Remaining running checks at review time: Build wn-agent and terminal harnesses (darwin-aarch64), Build wn-agent and terminal harnesses (darwin-x86_64), Rust tests (1/4), Simulator and vectors. GitHub still reports `BLOCKED`; merge remains contingent on the required CI gate passing.

**Review by Datawav — APPROVED** [review]
